File: apps/apple/Packages/AtelierShared/Sources/AtelierCore/WorkspaceItem.swift
import Foundation

public struct WorkspaceItem: Identifiable, Codable, Equatable, Sendable {
    public let id: UUID
    public var title: String
    public var summary: String
    public var markdown: String
    public var updatedAt: Date
    public var sectionID: String

    public init(
        id: UUID = UUID(),
        title: String,
        summary: String,
        markdown: String,
        updatedAt: Date = Date(),
        sectionID: String
    ) {
        self.id = id
        self.title = title
        self.summary = summary
        self.markdown = markdown
        self.updatedAt = updatedAt
        self.sectionID = sectionID
    }
}

File: apps/android/core/src/main/kotlin/diy/atelier/core/WorkspaceItem.kt
package diy.atelier.core

import java.time.Instant
import java.util.UUID

data class WorkspaceItem(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val summary: String,
    val markdown: String,
    val updatedAt: Instant = Instant.now(),
    val sectionId: String,
)
